### PR TITLE
fix: adds aria label to datagrid toolbar

### PR DESCRIPTION
**Pull request by davidmenendez**
Contributes to #4057

### CODE REVIEW

File: packages/ibm-products/src/components/Datagrid/Datagrid/addons/stateReducer.js [hunk 1]
```diff
@@ -114,7 +114,10 @@ export const stateReducer = (newState, action) => {
         const newData = { ...newState.selectedRowData };
         const dataWithRemovedRow = Object.fromEntries(
           Object.entries(newData).filter(([key]) => {
-            return parseInt(key) !== parseInt(getRowId(rowData.original, rowData.index));
+            return (
+              parseInt(key) !==
+              parseInt(getRowId(rowData.original, rowData.index))
+            );
           })
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Datagrid/Datagrid/addons/stateReducer.js:121]: I'm not seeing any real changes. It looks like you saved this page, and it reformatted it for you.

File: packages/ibm-products/src/components/Datagrid/utils/makeData.js [hunk 2]
```diff
@@ -24,8 +24,8 @@ const getRandomInteger = (min, max, decimalPlaces) => {
 };
 
 export const makeData = (...lens) => {
-  const config = lens.filter(a => typeof a === 'object')[0];
-  const filteredData = lens.filter(a => typeof a === 'number');
+  const config = lens.filter((a) => typeof a === 'object')[0];
+  const filteredData = lens.filter((a) => typeof a === 'number');
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Datagrid/utils/makeData.js:28]: I'm not seeing any real changes. It looks like you saved this page, and it reformatted it for you.

File: packages/ibm-products/src/components/Datagrid/Datagrid.stories.js [hunk 3]
```diff
@@ -496,7 +496,12 @@ export const BatchActions = () => {
     useStickyColumn
   );
 
-  return <Datagrid datagridState={{ ...datagridState }} />;
+  return (
+    <Datagrid
+      datagridState={{ ...datagridState }}
+      toolbarLabel="batch actions toolbar"
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Datagrid/Datagrid.stories.js:502]: Just a nit. This "label" will never be visible to the user. Should it be renamed to something more self-descriptive, like "toolbarAriaLabel"?

Comment (davidmenendez) [reply packages/ibm-products/src/components/Datagrid/Datagrid.stories.js]: fair enough. fixt

File: packages/ibm-products/src/components/Datagrid/utils/makeData.js (hunk 2 above)
Comment (davidmenendez) [reply packages/ibm-products/src/components/Datagrid/utils/makeData.js:28]: correct

**Review by paul-balchin-ibm — APPROVED** [review]

**Pull request merged**